src/adapter/src/catalog/storage.rs

```diff
@@ -1333,6 +1337,25 @@ impl<'a, S: Append> Transaction<'a, S> {
         }
     }
 
+    /// Removes all items in `ids` from the transaction.
+    ///
+    /// NOTE: On error, there still may be some items removed from the transaction. It is
+    /// up to the called to either abort the transaction or commit.
+    pub fn remove_items(&mut self, ids: BTreeSet<GlobalId>) -> Result<(), Error> {
```

Comment:
Why btree over hashset?

Reply:
No good reason to be honest. The only benefit is consistent error messages. Should I just convert to `HashSet`?

Reply:
That's a sufficient reason, keep it.